src/controllers/resources.js

```diff
@@ -544,7 +544,14 @@ export default function () {
           const getDoc = () => {
             if (i === ids.length) {
               // download CSV
-              const csv = Papa.unparse(exportData)
+              const header = Object.keys(exportData.reduce(function (result, obj) {
+                return Object.assign(result, obj)
+              }, {}))
+              const objHeader = header.reduce((obj, key) => (obj[key] = '', obj), {})
```

Comment:
Achei sua implementação aqui bem complexa mano, você percorrer arrays mais vezes que o necessário e vai dar muito `Object.assign` que não é uma operação barata, em backend não rola, aqui ainda vái que a CPU do usuário que tá pagando :stuck_out_tongue: 

Mas o que eu tinha mencionado por alto com você era até mais trivial na verdade, só algo como:

```suggestion
              const objHeader = {}
              exportData.forEach(row => {
                Object.keys(row).forEach(field => {
                  if (objHeader[field] === undefined) {
                    objHeader[field] = ''
                  }
                })
              })
```